### ISSUE TITLE
apparently subscription creation is working not consistently

### ISSUE THREAD

**Issue by jmcanterafonseca** [telefonicaid/fiware-orion#2184]
POST

``` json
{
    "subject": {
        "entities": [{
          "id": ""  
        }],
        "condition": {
            "attrs": ["capacity"]
        }
    },
    "notification": {
        "http": {
            "url": "http://localhost:8000"
        }
    }
}
```

Despite id is empty string the notification is created

If the notification is listed then the following result is obtained : idPattern .\* (I didn't intend to do so, and surprisingly timesSent 2, and I didn't update anything to make a notification happen. 

``` json
[
  {
    "id": "574425372be15a88e7be034a",
    "status": "active",
    "subject": {
      "entities": [
        {
          "id": "",
          "idPattern": ".*",
          "type": ""
        }
      ],
      "condition": {
        "attrs": [
          "capacity"
        ]
      }
    },
    "notification": {
      "timesSent": 2,
      "lastNotification": "2016-05-24T09:56:07.00Z",
      "attrs": [],
      "http": {
        "url": "http://localhost:8000"
      }
    }
  }
]
```

**Comment by fgalan** [comment]
Probably this is related with all the issues we already have in _"NGSIv2SubscripotionValidation" milestone, so I'm moving to that milestone this issue.

**Comment by jmcanterafonseca** [comment]
just to note that `timesSent` happens to be 2, even If I create the entity with a correct id. so I think here there are two bugs, empty id notifications which does not result in an error and timesSent and lastNotification reporting a notification without actually shouldn't have happened.

**Comment by crbrox** [comment]
Empty id/idPattern notifications should not be valid  any more. `timeSent` has been impossible to reproduce (maybe there were 2 entities which match the  "ALL" subscription , idPattern ".*"??)

To be closed if @jmcanterafonseca  agrees.

**Comment by fgalan** [comment]
For the records, it seems this was fixed in 1.3.0 (look to commit https://github.com/telefonicaid/fiware-orion/pull/2262/commits). Thus, milestone is set to that release.

However, the .test covering the case has been added in PR  https://github.com/telefonicaid/fiware-orion/pull/2816 during the 1.7.0 development cycle.
